src/lang/base/BuiltIns.mli

```diff
@@ -34,7 +34,7 @@ module ScillaBuiltIns
      * Executor for evaluating the operation      
     *)
     val find_builtin_op :
-      ER.rep tagged_builtin -> typ list -> ((typ * typ * built_in_executor), scilla_error list) result
+      builtin * ER.rep -> typ list -> ((typ * typ * built_in_executor), scilla_error list) result
```

Comment:
Here, `builtin * ER.rep` should be changed to `builtin_annot`.

Reply:
@jjcnn when I tried that I got
```
File "src/lang/base/BuiltIns.mli", line 37, characters 6-19:
Error: Unbound type constructor builtin_annot
```
I also tried `ScillaSyntax.builtin_annot` and got
```
Error: The module ScillaSyntax is a functor, not a structure
```

Reply:
Oh, right - you need the instantiated version of `ScillaSyntax`.

I think it should work if you add the line `module BISyntax = ScillaSyntax (SR) (ER)` at the beginning of the `ScillaBuiltIns` functor, and then change the signature of `find_builtin_op` to `BISyntax.builtin_annot -> ...`.

I could be mistaken, though. If you can't make it work, let me know and we'll merge it with `builtin * ER.rep -> ...` instead.

Reply:
@charles-cooper if you make `builtin_annot` a top-level definition, kind of like `'rep ident`, which is also top-level:
```
type 'rep builtin_annot = builtin * 'rep
[@@deriving sexp]
```
then it should pass through.

Reply:
Yeah, moving `builtin_annot` back outside the `ScillaSyntax` functor is probably the best option.

Reply:
Thanks for the suggestions